src/vmm/src/builder.rs

```diff
@@ -763,39 +767,572 @@ pub mod tests {
+            )
+        );
+
+        let err = MicroVMAlreadyRunning;
+        assert_eq!(format!("{}", err), "Microvm already running.");
+
+        let err = MissingKernelConfig;
+        assert_eq!(
+            format!("{}", err),
+            "Cannot start microvm without kernel configuration."
+        );
+
+        let err = NetDeviceNotConfigured;
+        assert_eq!(
+            format!("{}", err),
+            "The net device configuration is missing the tap device."
+        );
+
+        let err = OpenBlockDevice(io::Error::from_raw_os_error(0));
+        let mut inner_err_msg = format!("{:?}", io::Error::from_raw_os_error(0));
+        inner_err_msg = inner_err_msg.replace("\"", "");
+        assert_eq!(
+            format!("{}", err),
+            format!(
+                "Cannot open the block device backing file. {}",
+                inner_err_msg
+            )
+        );
+
+        let err = RegisterBlockDevice(device_manager::mmio::Error::EventFd(
+            io::Error::from_raw_os_error(0),
+        ));
+        assert_eq!(
+            format!("{}", err),
+            format!(
+                "Cannot initialize a MMIO Block Device or add a device to the MMIO Bus. {}",
+                device_manager::mmio::Error::EventFd(io::Error::from_raw_os_error(0))
+            )
+        );
+
+        let err = RegisterEvent(EventManagerError::EpollCreate(
+            io::Error::from_raw_os_error(0),
+        ));
+        assert_eq!(
+            format!("{}", err),
+            format!(
+                "Cannot register EventHandler. {:?}",
+                EventManagerError::EpollCreate(io::Error::from_raw_os_error(0))
+            )
+        );
+
+        let err = RegisterNetDevice(device_manager::mmio::Error::EventFd(
+            io::Error::from_raw_os_error(0),
+        ));
+        assert_eq!(
+            format!("{}", err),
+            format!(
+                "Cannot initialize a MMIO Network Device or add a device to the MMIO Bus. {}",
+                device_manager::mmio::Error::EventFd(io::Error::from_raw_os_error(0))
+            )
+        );
+
+        let err = RegisterVsockDevice(device_manager::mmio::Error::EventFd(
+            io::Error::from_raw_os_error(0),
+        ));
+        assert_eq!(
+            format!("{}", err),
+            format!(
+                "Cannot initialize a MMIO Vsock Device or add a device to the MMIO Bus. {}",
+                device_manager::mmio::Error::EventFd(io::Error::from_raw_os_error(0))
+            )
+        );
+    }
+
+    #[test]
+    fn test_kernel_cmdline_err_to_startuvm_err() {
+        let err = StartMicrovmError::from(kernel::cmdline::Error::HasSpace);
+        assert_eq!(
+            format!("{}", err),
+            format!(
+                "Invalid kernel command line: {}",
+                kernel::cmdline::Error::HasSpace.to_string()
+            )
+        );
     }
+
+    // TODO: implement the test after vsock refactoring PR is merged
```

Comment:
Should we add them before merging this since vsock is merged?

Reply:
Done, please take a look.